tests/dataloader_tests.cpp

```diff
@@ -9,74 +9,77 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-#define BOOST_TEST_DYN_LINK
-#define BOOST_TEST_MAIN // Do not define this anywhere else.
-#include <dataloader/dataloader.hpp>
-#include <boost/test/unit_test.hpp>
 
-using namespace boost::unit_test;
+/**
+ * CATCH_CONFIG_MAIN tells Catch to provide a main() don't include
+ * this in other test files.
+ */
+#define CATCH_CONFIG_MAIN
```

Comment:
I'd suggest defining `main()` ourselves so that we can, e.g., set the random seed.  Take a look at how it's done in ensmallen: https://github.com/mlpack/ensmallen/blob/master/tests/main.cpp

Reply:
I have applied this suggestion, have a look.